pyCGM_Single/tests/test_angle_calculations.py

```diff
@@ -0,0 +1,31 @@
+import unittest
+import pyCGM_Single.pyCGM as pyCGM
+
+class TestCalculations(unittest.TestCase):
+
+    def test_getangle_sho(self):
+        axisP = [[0.09010104879445179904, 0.99590937599884910014, 0.00680557152145411237],
+                 [0.99377608791559168822, -0.08945440277921079542, -0.06638521605464120512],
+                 [-0.06550487076049194002, 0.01274459183355247660, -0.99777085910818641423]]
+        axisD = [[0.14362551354236074985, -0.98963110439615320502, -0.00141034441480769601],
+                 [0.68540404438920177199, 0.09844473917718232769, 0.72147760135931093828],
+                 [-0.71385783444225126004, -0.10458924677044478813, 0.69243633762630452111]]
+        self.assertEqual(pyCGM.getangle_sho(axisP, axisD), [-9.42569150165887, 130.86647058885387, -170.3887751198432])
```

Comment:
Lets assign the values first, then round the values to some precision (say, 8 units after the float), then do the assert check

Reply:
I got these values by running the example code and using the input and output values. I had to increase the precision on the input values as they wouldn't pass the assertion unless they had 18-20 decimal points. I can round the input values to 8, but then I would need to change the output values and they wouldn't be the same as the ones in the example code

Reply:
Don't change the input, change the rounding of the output. 

It would look something like this

```
    axisP = [[0.13232935635315357104, 0.98562945782504129966, -0.10499292030749529658],
                 [-0.99119134439010281312, 0.13101087562301927392, -0.01938734831355759525],
                 [-0.00535352718324588750, 0.10663358915485332545, 0.99428397221845443710]]
    axisD = [[0.09010104879445179904, 0.99590937599884910014, 0.00680557152145411237],
                 [0.99377608791559168822, -0.08945440277921079542, -0.06638521605464120512],
                 [-0.06550487076049194002, 0.01274459183355247660, -0.99777085910818641423]]

    result = pyCGM.getangle_sho(axisP, axisD)
    expected = [-9.42569150165887, 130.86647058885387, -170.3887751198432]
    self.assertEqual(np.around(result,decimals=8), np.around(expected, decimals=8))
```

Reply:
Although really the "8" should be a variable called ' rounding_precision' that we set for tests

Reply:
I updated the pull request, but after doing so I realized that np.testing.assert_almost_equal might be a good option. I believe it is doing the same thing as assertEqual and the rounding, the tests still pass, and I would assume it is more efficient as it is one function that serves the same purposes of the 3 currently. Do you think it’d be a good option here? Like:

`np.testing.assert_almost_equal(result, expected, rounding_precision)`

I also forgot to update it with the precision variable so I’ll do that at the same time

Reply:
For something simple like the assert equal I think it is fine, but we don't want to go down the route of using all of numpy testing features as it just makes things more convoluted later.

Reply:
Okay, sounds good. I updated the pull request. If it looks good I'll start working on the tests for the other functions in the file